lib/frodata/concerns/api.rb

```diff
@@ -0,0 +1,262 @@
+      #   client.api_put 'leads(073ca9c8-2a41-e911-a81d-000d3a1d5a0b)', { ... }
+      #
+      #   # Perform a delete request
+      #   client.delete '/api/data/v9.1/leads(073ca9c8-2a41-e911-a81d-000d3a1d5a0b)'
+      #   client.api_delete 'leads(073ca9c8-2a41-e911-a81d-000d3a1d5a0b)'
+      #
+      # Returns the Faraday::Response.
+      define_verbs :get, :post, :put, :delete, :patch, :head
+
+      def metadata
+        api_get("$metadata").body
+      end
+
+      # Public: Executs a SOQL query and returns the result.
+      #
+      # soql - A SOQL expression.
+      #
+      # Examples
+      #
+      #   # Find the names of all Accounts
+      #   client.query('select Name from Account').map(&:Name)
+      #   # => ['Foo Bar Inc.', 'Whizbang Corp']
+      #
+      # Returns a Restforce::Collection if Restforce.configuration.mashify is true.
+      # Returns an Array of Hash for each record in the result if
+      # Restforce.configuration.mashify is false.
+      def query(soql)
+        response = api_get 'query', q: soql
+        mashify? ? response.body : response.body['records']
+      end
+
+      # Public: Insert a new record.
+      #
+      # entity_set - The set the entity belongs to
+      # attrs   - Hash of attributes to set on the new record.
+      #
+      # Examples
+      #
+      #   # Add a new lead
+      #   client.create('leads', {"firstname" =>'Bob'})
+      #   # => '073ca9c8-2a41-e911-a81d-000d3a1d5a0b'
+      #
+      # Returns the String Guid of the newly created entity.
+      # Returns false if something bad happens.
+      def create(*args)
+        create!(*args)
+      rescue *exceptions
+        false
+      end
+      alias insert create
+
+      # Public: Insert a new record.
+      #
+      # entity_set - The set the entity belongs to
+      # attrs   - Hash of attributes to set on the new record.
+      #
+      # Examples
+      #
+      #   # Add a new lead
+      #   client.create!('leads', {"firstname" =>'Bob'})
+      #   # => '073ca9c8-2a41-e911-a81d-000d3a1d5a0b'
+      #
+      # Returns the String Guid of the newly created entity.
+      # Raises exceptions if an error is returned from Dynamics.
+      def create!(entity_set, attrs)
+        entity = service[entity_set].new_entity(attrs)
+        url_chunk = to_url_chunk(entity)
+        url = api_post(url_chunk, attrs).headers['odata-entityid']
+        id = url.match(/\(.+\)/)[0]
+      end
+      alias insert! create!
+
+      # Public: Update a record.
+      #
+      # entity_set - The set the entity belongs to
+      # attrs   - Hash of attributes to set on the record.
+      #
+      # Examples
+      #
+      #   # Update the lead with id '073ca9c8-2a41-e911-a81d-000d3a1d5a0b'
+      #   client.update('leads', "leadid": '073ca9c8-2a41-e911-a81d-000d3a1d5a0b', Name: 'Whizbang Corp')
+      #
+      # Returns true if the sobject was successfully updated.
```

Comment:
Nit: Remove subject

Reply:
yea was not done . was a draft PR. Pushing the update soon for a more final PR